pkg/envoy/cds/response.go

```diff
@@ -22,37 +22,28 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
 	// Github Issue #1575
 	proxyServiceName := svcList[0]
 
-	resp := &xds_discovery.DiscoveryResponse{
-		TypeUrl: string(envoy.TypeCDS),
-	}
 	// The clusters have to be unique, so use a map to prevent duplicates. Keys correspond to the cluster name.
-	clusterFactories := make(map[string]*xds_cluster.Cluster)
+	var clusters []*xds_cluster.Cluster
```

Comment:
The reason to use a map here was to avoid programming clusters with the same name, which would be rejected by Envoy. Could you retain this or ensure this never happens?

Reply:
Ideally this should be guaranteed by `ListAllowedOutboundServicesForIdentity()`
I'll put the uniqueness guarantee here again.

Reply:
I agree with you, however if a cluster internal to the control plane (Ex. prometheus, tracing) ends up using the same cluster name as an outbound service, or vice versa, this could break. I think with a map at least we can detect this within the controller and log an error. What do you think?

Reply:
Better be safe than sorry - I'll run deduplication at the end before marshaling!